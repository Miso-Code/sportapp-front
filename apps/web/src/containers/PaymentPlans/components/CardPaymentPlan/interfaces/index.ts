import { HTMLAttributes } from 'react'

export interface Props extends HTMLAttributes<HTMLDivElement> {
	name: string
	description: string
	price: number
	duration: string
	isActive?: boolean
	activeText?: string
	benefits: Benefits[]
	className?: string
}

export interface Benefits {
	description: string
	isActive: boolean
}
