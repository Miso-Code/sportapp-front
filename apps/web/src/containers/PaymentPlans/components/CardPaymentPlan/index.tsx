import CheckCircleIcon from '@mui/icons-material/CheckCircle'
import CancelIcon from '@mui/icons-material/Cancel'
import {
	Card,
	CardContent,
	Chip,
	List,
	ListItem,
	ListItemIcon,
	ListItemText,
	Typography
} from '@mui/material'
import { Props } from './interfaces'

export default function CardPaymentPlan({
	name,
	description,
	className = '',
	benefits,
	price,
	duration,
	isActive = false,
	activeText = 'Active',
	...props
}: Props) {
	return (
		<Card className={`px-2 pb-4 max-w-[24rem] ${className}`} {...props}>
			<CardContent className='flex justify-between'>
				<Typography variant='h6'>{name}</Typography>
				{isActive && (
					<Chip
						label={activeText}
						color='primary'
						variant='outlined'
						className='bg-[#2195f337]'
						size='medium'
					/>
				)}
			</CardContent>
			<CardContent className='py-0'>
				<Typography variant='caption'>{description}</Typography>
			</CardContent>
			<CardContent className='flex items-end'>
				<Typography component='p' variant='h2'>
					${price}
					<span className='text-3xl'>/ {duration}</span>
				</Typography>
			</CardContent>
			<CardContent className='py-0'>
				<List>
					{benefits.map((benefit, index) => (
						<ListItem key={index} className='p-0'>
							<ListItemIcon>
								{benefit.isActive ? (
									<CheckCircleIcon />
								) : (
									<CancelIcon />
								)}
							</ListItemIcon>
							<ListItemText primary={benefit.description} />
						</ListItem>
					))}
				</List>
			</CardContent>
		</Card>
	)
}
