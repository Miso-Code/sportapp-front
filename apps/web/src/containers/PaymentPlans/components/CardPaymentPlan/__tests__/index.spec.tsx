import CardPaymentPlan from '..'
import { render, RenderResult } from '@testing-library/react'

describe('CardPaymentPlan', () => {
	let wrapper: RenderResult

	beforeEach(() => {
		wrapper = render(
			<CardPaymentPlan
				benefits={[
					{
						description: 'description 1',
						isActive: true
					},
					{
						description: 'description 2',
						isActive: true
					},
					{
						description: 'description 3',
						isActive: true
					},
					{
						description: 'description 4',
						isActive: true
					},
					{
						description: 'description 5',
						isActive: true
					}
				]}
				duration='mes'
				price={5}
				description='description'
				name='name'
			/>
		)
	})

	it('should render the component', () => {
		expect(wrapper).toMatchSnapshot()
	})

	it('should render the component with the correct name', () => {
		expect(wrapper.getByText('name')).toBeInTheDocument()
	})

	it('should render the component with the correct description', () => {
		expect(wrapper.getByText('description')).toBeInTheDocument()
	})

	it('should render the component with the correct price', () => {
		expect(wrapper.getByText('$5')).toBeInTheDocument()
	})

	it('should render the component with the correct duration', () => {
		expect(wrapper.getByText('/ mes')).toBeInTheDocument()
	})

	it('should render the component with the correct benefits', () => {
		expect(wrapper.getByText('description 1')).toBeInTheDocument()
		expect(wrapper.getByText('description 2')).toBeInTheDocument()
		expect(wrapper.getByText('description 3')).toBeInTheDocument()
		expect(wrapper.getByText('description 4')).toBeInTheDocument()
		expect(wrapper.getByText('description 5')).toBeInTheDocument()
	})
})
