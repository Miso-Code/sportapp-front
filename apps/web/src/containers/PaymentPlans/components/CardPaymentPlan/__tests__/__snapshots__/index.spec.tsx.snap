// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentPlans should render the component 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root px-2 pb-4 max-w-[24rem]  css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root flex justify-between css-46bh2p-MuiCardContent-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
          >
            name
          </h6>
        </div>
        <div
          class="MuiCardContent-root py-0 css-46bh2p-MuiCardContent-root"
        >
          <span
            class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
          >
            description
          </span>
        </div>
        <div
          class="MuiCardContent-root flex items-end css-46bh2p-MuiCardContent-root"
        >
          <p
            class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
          >
            $
            5
            <span
              class="text-3xl"
            >
              / 
              mes
            </span>
          </p>
        </div>
        <div
          class="MuiCardContent-root py-0 css-46bh2p-MuiCardContent-root"
        >
          <ul
            class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
          >
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  description 1
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  description 2
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  description 3
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  description 4
                </span>
              </div>
            </li>
            <li
              class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
            >
              <div
                class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="CheckCircleIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                  />
                </svg>
              </div>
              <div
                class="MuiListItemText-root css-tlelie-MuiListItemText-root"
              >
                <span
                  class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
                >
                  description 5
                </span>
              </div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root px-2 pb-4 max-w-[24rem]  css-bhp9pd-MuiPaper-root-MuiCard-root"
    >
      <div
        class="MuiCardContent-root flex justify-between css-46bh2p-MuiCardContent-root"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6 css-2ulfj5-MuiTypography-root"
        >
          name
        </h6>
      </div>
      <div
        class="MuiCardContent-root py-0 css-46bh2p-MuiCardContent-root"
      >
        <span
          class="MuiTypography-root MuiTypography-caption css-1sn4lm3-MuiTypography-root"
        >
          description
        </span>
      </div>
      <div
        class="MuiCardContent-root flex items-end css-46bh2p-MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-h2 css-1sra7t5-MuiTypography-root"
        >
          $
          5
          <span
            class="text-3xl"
          >
            / 
            mes
          </span>
        </p>
      </div>
      <div
        class="MuiCardContent-root py-0 css-46bh2p-MuiCardContent-root"
      >
        <ul
          class="MuiList-root MuiList-padding css-h4y409-MuiList-root"
        >
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                description 1
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                description 2
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                description 3
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                description 4
              </span>
            </div>
          </li>
          <li
            class="MuiListItem-root MuiListItem-gutters MuiListItem-padding p-0 css-1p823my-MuiListItem-root"
          >
            <div
              class="MuiListItemIcon-root css-cveggr-MuiListItemIcon-root"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="CheckCircleIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8z"
                />
              </svg>
            </div>
            <div
              class="MuiListItemText-root css-tlelie-MuiListItemText-root"
            >
              <span
                class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
              >
                description 5
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
